src/metabase/integrations/slack.clj

```diff
@@ -117,16 +117,15 @@
   true)
 
 (defn- handle-error [body]
-  (let [invalid-token? (slack-token-error-codes (:error body))
-        message        (if invalid-token?
-                         "Invalid token"
-                         (format "Slack API error: %s" (:error body)))
-        error          (if invalid-token?
-                         {:error-code (:error body)
-                          :errors     {:slack-token message}}
-                         {:error-code (:error body)
-                          :message    message
-                          :response   body})]
+  (let [invalid-token?   (slack-token-error-codes (:error body))
+        missing-channel? (= (:error body) "channel_not_found")
+        message          (format "Slack API error: %s" (:error body))
+        error-stub       {:error-code (:error body)
+                          :error-type :slack/unknown}
+        error            (cond
+                           invalid-token?   (assoc error-stub :error-type :slack/invalid-token)
+                           missing-channel? (assoc error-stub :error-type :slack/channel-not-found)
+                           :else            (assoc error-stub :response body))]
```

Comment:
another way to shorten this cond is to use `cond->`


```clojure
(cond-> {:error-code (:error body)
                           :error-type :slack/unknown}
                     invalid-token?   (assoc :error-type :slack/invalid-token)
                     missing-channel? (assoc :error-type :slack/channel-not-found)
                     true            (assoc :response body))
```

Reply:
Thanks. I'm new to Clojure, so any tips you have are welcome.

Reply:
Oh, but this form does not short circuit so the `:response` will always be in the map. Is that something we want?